src/thunderbots/software/ai/navigator/obstacle/velocity_obstacle.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "ai/navigator/obstacle/obstacle.h"
+#include "shared/constants.h"
+
+// Placeholder obstacle represents a stationary robot centred at (0,0)
+class VelocityObstacle : public Obstacle
+{
+   public:
+    explicit VelocityObstacle();
+
+    /*
+     * Gets the boundary polygon around the given robot obstacle that other robots
+     * should not enter with a buffer scaled by radiusScaling and
+     * velocity_projection_scaling
+     *
+     * @param robot robot to create obstacle boundary polygon around
+     * @param radiusScaling safety factor to vary the radius size of the buffer zone
+     * (1=default), must be greater than 0
```

Comment:
could we make this the buffer size in meters? When numbers like this don't have units it can be very difficult to reason about what they mean, what are reasonable values to use for them, etc.

Reply:
I see why Jon used a scale factor here. If we go with a defined unit of measurement we would have to switch to a constant that is added to the margins.

Depends what we are trying to achieve with this.

Reply:
Yea, but I think a buffer obfuscates things as well. The caller has to "know" what the base value being scaled is and so doesn't necessarily have context to provide a "good" scaling value.

I'm not quire sure about what you mean regarding a constant being added to the margins? This "constant" would just be a parameter of how much to add to the obstacle buffer, which would represent how much to avoid an obstacle by? (ie. the distance between the edge of the robot and the obstacle)

Reply:
Yeah exactly that. 

If we use multiplication for scaling everything is relative; there's a reason why all safety factors in engineering are multiplicative and not a constant.

If a ball is moving at 5m/s and we avoid by 10cm that doesn't leave much room for error.

It's definitely a trade-off, by using a constant value the avoid-distance is more knowingly specified by the user. The user will have to be more conscious of the situation the function is being called for. 
Which necessarily isn't a bad thing.

Reply:
Going back and reading my original comment, I realize I didn't convey my thoughts very well. I don't have anything against multiplicative constants, I just want to make sure that a user could understand the physical interpretation of what the factor they use is. For example, something like the following would be totally fine with me:
```
@param robot_radius_scaling The percentage by which the robot radius will be inflated by (ie. 0 would be no inflation, 1 would double the robot radius)
```

The reason that I suggested adding a constant with a defined unit of measurement is that I think it's a little easier to reason about, and it's fairly trivial for the user of this function to multiply it by whatever factor they think is important. I'm totally chill with either option here, just as long as the function interface is clear.

Reply:
Ok I'll make the interface clear when I rewrite this for the new obstacle architecture.